### PR TITLE
46 need to import yolo format dataset

### PR DESCRIPTION
**Pull request by oneQuery**
- Added importing yolo format dataset

### CODE REVIEW

**Review by echoyunjin — DISMISSED** [review]
If only the format of coco and yolo is different, it seems that abstraction can be done for scalability, but I don't know what to do now. First of all, I think it would be good to think about it together with Merge.

**Comment by ZeroAct** [comment]
please resolve conflicts.

**Comment by oneQuery** [comment]
> please resolve conflicts.

@ZeroAct done

File: waffle_utils/dataset/dataset.py [hunk 1]
```diff
@@ -204,16 +206,16 @@ def load(cls, name: str, root_dir: str = None) -> "Dataset":
     def from_coco(
         cls,
         name: str,
-        coco_file: str,
-        coco_root_dir: str,
```
Comment (ZeroAct) [inline waffle_utils/dataset/dataset.py:208]: why did it changed?

File: waffle_utils/dataset/dataset.py [hunk 2]
```diff
@@ -256,7 +258,87 @@ def from_coco(
             )
 
         # copy raw images
-        io.copy_files_to_directory(coco_root_dir, ds.raw_image_dir)
+        io.copy_files_to_directory(images_dir, ds.raw_image_dir)
+
+        return ds
+
+    @classmethod
+    def from_yolo(
```
Comment (ZeroAct) [inline waffle_utils/dataset/dataset.py:266]: this function only supports object_detection. check the classification format.

File: waffle_utils/dataset/dataset.py [hunk 3]
```diff
@@ -791,3 +873,136 @@ def _export(images: list[Image], set_name: str, export_dir: Path):
...
+                    },
+                    "area": area,
+                }
+            )
+
+            ds.add_annotations([annotation])
+            annotation_id += 1
+
+        # Return the updated annotation_id
+        return annotation_id
```
Comment (ZeroAct) [inline waffle_utils/dataset/dataset.py:1008]: These function shouldn't be class method. As we talked about (before)[https://github.com/snuailab/waffle_utils/pull/52#discussion_r1147249190], these should be moved to the `from_yolo` function.

File: waffle_utils/dataset/dataset.py (hunk 1 above)
Comment (oneQuery) [reply waffle_utils/dataset/dataset.py:208]: The name `coco` is more like format type, not the image type. So it's confusing what the `coco_root` means. It looks like the `coco_root_dir` represents the dataset root dir, which already exists as `root_dir`. The `coco_root_dir` here is actually the images directory which is independent to the format type. 
```
# mnist sample dataset directories
dataset/  # `root_dir`
    ├ images/  # `coco_root_dir`? 🤔
    │   ├ 1.png
    │   ├ 2.png
    │   └ ...
    └ coco.json  # `coco_file`
```

File: waffle_utils/dataset/dataset.py (hunk 2 above)
Comment (ZeroAct) [reply waffle_utils/dataset/dataset.py:266]: I think we should add `task` in `Dataset` class.
😂😂😂😂

File: waffle_utils/run.py [hunk 4]
```diff
@@ -125,7 +156,9 @@ def _extract_frames(
 ):
     """Extract Frames from a Video File"""
 
-    extract_frames(input_path, output_dir, frame_rate, output_image_extension, verbose)
+    extract_frames(
+        input_path, output_dir, frame_rate, output_image_extension, verbose
+    )
```
Comment (dustjr1565) [inline waffle_utils/run.py:161]: Has the contents of .pre-commit-config.yaml changed recently? 
It looks like it has been changed even though no new code was added

File: waffle_utils/dataset/dataset.py (hunk 1 above)
Comment (ZeroAct) [reply waffle_utils/dataset/dataset.py:208]: I agree. 
These types of changes should be notified to TX team and need to update our documentation.
Say it loud!

File: waffle_utils/dataset/dataset.py (hunk 2 above)
Comment (oneQuery) [reply waffle_utils/dataset/dataset.py:266]: What you mean is, separating how the `from_yolo()` imports the dataset according to the task? Such as like this?
```python
# waffle_utils/dataset/dataset.py
class Dataset:
    ...
    def get_task(...):
        ...
       return task

    def from_yolo(...):
        task = get_task()
        if task == 'classification':
            ...
        elif task == 'object_detection':
            ...
        else:
            ...
```

File: waffle_utils/run.py (hunk 4 above)
Comment (oneQuery) [reply waffle_utils/run.py:161]: This is my current `.pre-commit-config.yaml`. I have never changed it before.
```yaml
# Define hooks for code formations
# Will be applied on any updated commit files if a user has installed and linked commit hook

default_language_version:
  python: python3.9

exclude: 'docs/'
# Define bot property if installed via https://github.com/marketplace/pre-commit-ci
ci:
  autofix_prs: true
  autoupdate_commit_msg: '[pre-commit.ci] pre-commit suggestions'
  autoupdate_schedule: monthly
  # submodules: true

repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.4.0
    hooks:
      # - id: end-of-file-fixer
      - id: trailing-whitespace
      - id: check-case-conflict
      - id: check-yaml
      - id: check-toml
      - id: pretty-format-json
        args:
          - --autofix
      - id: check-docstring-first

  - repo: https://github.com/asottile/pyupgrade
    rev: v3.3.1
    hooks:
      - id: pyupgrade
        name: Upgrade code
        args: [ --py37-plus ]

  - repo: https://github.com/pycqa/isort
    rev: 5.12.0
    hooks:
      - id: isort
        name: Sort imports
        args: [
          --multi-line=3,
          --split-on-trailing-comma
          ]

  - repo: https://github.com/psf/black
    rev: 22.12.0
    hooks:
    - id: black
      args: [--line-length=79]

  # - repo: https://github.com/PyCQA/flake8
  #   rev: 6.0.0
  #   hooks:
  #     - id: flake8
  #       name: PEP8
  #       args: ["--ignore=E501,E741"]
```

File: waffle_utils/dataset/dataset.py (hunk 2 above)
Comment (oneQuery) [reply waffle_utils/dataset/dataset.py:266]: Ah, into the property. now I get it.

**Review by dustjr1565 — APPROVED** [review]

**Comment by oneQuery** [comment]
@dustjr1565 Hey.,. Don't approve anything without review...

**Comment by ZeroAct** [comment]
@dustjr1565 
re pull request this issue after you've done.

**Pull request closed without merging**